pkg/stanza/operator/input/windows/input.go

```diff
@@ -197,8 +198,11 @@ func (i *Input) readOnInterval(ctx context.Context) {
 
 // read will read events from the subscription.
 func (i *Input) read(ctx context.Context) {
-	events, err := i.subscription.Read(i.maxReads)
-	if err != nil {
+	events, err := i.subscription.Read(i.currentMaxReads)
+	if errors.Is(err, ErrBatchSizeReduced) {
+		i.currentMaxReads = max(i.currentMaxReads/2, 1)
+		i.Logger().Debug("Encountered RPC_S_INVALID_BOUND, reducing batch size", zap.Int("current batch size", i.currentMaxReads), zap.Int("original batch size", i.maxReads))
```

Comment:
As pointed before returning an error but using the data is not typical in golang.